lte/gateway/c/core/oai/tasks/gtpv1-u/ebpf.h

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2022 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* File : ebpf.h
+ */
+
+#pragma once
+
+#include <unistd.h>
+#include <linux/bpf.h>
+
+#ifndef __NR_bpf
+#if defined(__i386__)
+#define __NR_bpf 357
+#elif defined(__x86_64__)
+#define __NR_bpf 321
+#elif defined(__aarch64__)
+#define __NR_bpf 280
+#elif defined(__sparc__)
+#define __NR_bpf 349
+#elif defined(__s390__)
+#define __NR_bpf 351
+#else
+#error __NR_bpf not defined. libbpf does not support your arch.
+#endif
+#endif
+
+#define BPF_ANY 0     /* create new element or update existing */
+#define BPF_NOEXIST 1 /* create new element only if it didn't exist */
+#define BPF_EXIST 2   /* only update existing element */
+
+#define DL_MAP_PATH "/sys/fs/bpf/dl_map"
+
+static unsigned long ptr_to_u64(const void* ptr) {
+  return (unsigned long) ptr;
+}
+
+static inline long sys_bpf(
+    enum bpf_cmd cmd, union bpf_attr* attr, unsigned int size) {
+  return syscall(__NR_bpf, cmd, attr, size);
+}
+
+long bpf_obj_get(const char* pathname);
+long bpf_map_update_elem(
+    unsigned int fd, void* key, void* value, unsigned long long flags);
+long bpf_map_lookup_elem(unsigned int fd, void* key, void* value);
+
+long bpf_obj_get(const char* pathname) {
+  union bpf_attr attr;
+
+  bzero(&attr, sizeof(attr));
+  attr.pathname = ptr_to_u64((const void*) pathname);
+
+  return sys_bpf(BPF_OBJ_GET, &attr, sizeof(attr));
+}
+
+long bpf_map_update_elem(
+    unsigned int fd, void* key, void* value, unsigned long long flags) {
+  union bpf_attr attr;
+
+  bzero(&attr, sizeof(attr));
+  attr.map_fd = fd;
+  attr.key    = ptr_to_u64(key);
+  attr.value  = ptr_to_u64(value);
+  attr.flags  = flags;
+
+  return sys_bpf(BPF_MAP_UPDATE_ELEM, &attr, sizeof(attr));
+}
+
+long bpf_map_delete_elem(unsigned int fd, void* key) {
+  union bpf_attr attr;
+  int ret;
+
+  memset(&attr, 0, sizeof(attr));
+  attr.map_fd = fd;
+  attr.key    = ptr_to_u64(key);
+
+  return sys_bpf(BPF_MAP_DELETE_ELEM, &attr, sizeof(attr));
+}
+
+struct bpf_map_val {
+  uint32_t ip;
+  uint32_t tei;
+};
+
+void add_ebpf_dl_map_entry(
```

Comment:
Lets define separate lib for ebpf helper functions and DL map specific functions.